onnxruntime/core/optimizer/layer_norm_fusion.cc

```diff
@@ -66,6 +76,22 @@ Status LayerNormFusion::ApplyImpl(Graph& graph, bool& modified, int graph_level,
         !IsSupportedDataType(reduce_mean_node)) {
       continue;
     }
+
+    // verify axes is contiguous if more than one elements
+    const onnxruntime::NodeAttributes& attributes = reduce_mean_node.GetAttributes();
+    std::vector<int64_t> axes_values;
+    if (attributes.find("axes") != attributes.end()) {
+      axes_values = RetrieveValues<int64_t>(attributes.at("axes"));
+      if (axes_values.size() > 1) {
+        sort(axes_values.begin(), axes_values.end());
+        for (size_t i = 0; i < axes_values.size() - 1; i++) {
+          if (axes_values[i] + 1 != axes_values[i + 1]) {
+            continue;
```

Comment:
Same as the other comment. the `continue` might not work as intended.

Reply:
good catch

---
In reply to: [444568745](https://github.com/microsoft/onnxruntime/pull/4320#discussion_r444568745) [](ancestors = 444568745)